src/main/java/frc/robot/subsystems/climb/Climb.java

```diff
@@ -0,0 +1,220 @@
+        }
+
+        if(_rightHeight != null)
+        {
+            _io.setAlgorithmVoltageRight(_rightPID.calculate(getExtensionRight(), _rightHeight));
+        }
+    }
+
+    public void stop()
+    {
+        _io.setVoltageLeft(0.0);
+        _io.setVoltageRight(0.0);
+
+        _climbSetpointLeft  = null;
+        _climbSetpointRight = null;
+
+        setLockState(true);
+    }
+
+    public void setLockState(boolean enabled)
+    {
+        _io.setLockStateLeft(enabled, _inputs);
+        _io.setLockStateRight(enabled, _inputs);
+    }
+
+    public void setSetpointLeft(double setpoint)
+    {
+        _climbSetpointLeft = MathUtil.clamp(setpoint, _climbMinExtension, _climbMaxExtension);
+    }
+    
+    public void setSetpointRight(double setpoint)
+    {
+        _climbSetpointRight = MathUtil.clamp(setpoint, _climbMinExtension, _climbMaxExtension);
+    }
+
+    public void setVoltageLeft(double volts)
+    {
+        _climbSetpointLeft = null;
+        
+        _io.setLockStateLeft(Math.abs(volts) <= 0.12, _inputs); // TODO: Change number to constants
+        
+        _io.setVoltageLeft(volts);
+    }
+
+    public void setVoltageRight(double volts)
+    {
+        _climbSetpointRight = null;
+        
+        _io.setLockStateRight(Math.abs(volts) <= 0.12, _inputs); // TODO: Change number to constants
+        
+        _io.setVoltageRight(volts);
+    }
+    
+    public boolean isAtLeftSetpoint()
+    {
+        return _climbFeedbackLeft.atSetpoint();
+    }
+    
+    public boolean isAtRightSetpoint()
+    {
+        return _climbFeedbackRight.atSetpoint();
+    }
+
+    public void setExtensionMax(double max)
+    {
+        _climbMaxExtension = max;
+    }
+
+    public void setExtensionMin(double min)
+    {
+        _climbMinExtension = min;
+    }
+    
+    // Mainly for dashboard
+    public double getExtensionLeft()
+    {
+        return _inputs.extensionLeft;
+    }
+    
+    // Mainly for dashboard
+    public double getExtensionRight()
+    {
+        return _inputs.extensionRight;
+    }
+
+    public void algorithmStop()
+    {
+        _io.setAlgorithmVoltageLeft(0.0);
+        _io.setAlgorithmVoltageRight(0.0);
+    }
+    
+    public void setAlgorithmSetpointLeft(double leftHeight)
+    {
+        _leftHeight = leftHeight;
+    }
+    
+    public void setAlgorithmSetpointRight(double rightHeight)
+    {
+        _rightHeight = rightHeight;
+    }
+    
+    public void setAlgorithmVoltageLeft(double volts)
+    {
+        _leftHeight = null;
+        
+        _io.setAlgorithmVoltageLeft(volts);
+    }
+    
+    public void setAlgorithmVoltageRight(double volts)
+    {
+        _leftHeight = null;
+        
+        _io.setAlgorithmVoltageRight(volts);
+    }
+
+    public boolean isAtLeftSetpointAlgorithm()
+    {
+        return _leftPID.atSetpoint();
+    }
+
+    public boolean isAtRightSetpointAlgorithm()
+    {
+        return _rightPID.atSetpoint();
+    }
```

Comment:
Each of these methods can be removed. We have equivalent non-algorithm methods, and there's not really a need for us to keep the algorithm and non-algorithm logic separate